File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.1/JMBA_Kiosk_iOS/JambaKiosk/AuditUser.swift
//
//  AuditUser.swift
//  JambaJuice
//
//  Created by Eneko Alonso on 8/31/15.
//  Copyright (c) 2015 Jamba Juice. All rights reserved.
//

import Foundation
import Parse

struct AuditUser {

    static let parseClassName = "AuditUser"

    private var user: User
    private var action: String

    init(user: User, action: String) {
        self.user = user
        self.action = action
    }

    func serializeAsParseObject() -> PFObject {
        let parseObject = PFObject(className: AuditUser.parseClassName)
        parseObject["id"]          = user.userId ?? NSNull()
        parseObject["email"]       = user.emailAddress ?? NSNull()
        parseObject["phone"]       = user.phoneNumber ?? NSNull()
        parseObject["firstName"]   = user.firstName ?? NSNull()
        parseObject["lastName"]    = user.lastName ?? NSNull()
        parseObject["dateOfBirth"] = NSNull()
        parseObject["smsOptIn"]    = NSNull()
        parseObject["emailOptIn"]  = NSNull()
        parseObject["action"]      = action
        return parseObject
    }

}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-jamba-development/JMBA_JambaJuice_Android/app/src/main/java/com/olo/jambajuice/BusinessLogic/Models/AuditUser.java
package com.olo.jambajuice.BusinessLogic.Models;

import com.olo.jambajuice.Utils.ParseUtils;
import com.parse.ParseObject;

/**
 * Created by Nauman Afzaal on 01/09/15.
 */
public class AuditUser
{
    public static String parseClassName = "AuditUser";
    User user;
    String action;

    public AuditUser(User user, String action)
    {
        this.user = user;
        this.action = action;
    }

    public ParseObject serializeAsParseObject()
    {
        ParseObject parseObject = ParseObject.create(AuditUser.parseClassName);
        parseObject.put("id", ParseUtils.sanitizeValue(user.getSpendGoId()));
        parseObject.put("email", ParseUtils.sanitizeValue(user.getEmailaddress()));
        parseObject.put("phone", ParseUtils.sanitizeValue(user.getContactnumber()));
        parseObject.put("firstName", ParseUtils.sanitizeValue(user.getFirstname()));
        parseObject.put("lastName", ParseUtils.sanitizeValue(user.getLastname()));
        parseObject.put("dateOfBirth",ParseUtils.sanitizeValue(user.getDob()));
        parseObject.put("smsOptIn", ParseUtils.sanitizeValue(user.isEnableSmsOpt()));
        parseObject.put("emailOptIn", ParseUtils.sanitizeValue(user.isEnableEmailOpt()));
        parseObject.put("action", ParseUtils.sanitizeValue(action));
        return parseObject;
    }
}
